crates/wash-cli/src/cmd/wit/deps.rs

```diff
@@ -0,0 +1,321 @@
+            &wasmcloud_toml_dir,
+            &mut wkg_config_overrides,
+            &mut wkg_client_config,
+        )
+        .await?;
+    }
+
+    // Re-add the modified local wkg config overrides
+    wkg_config.overrides = Some(wkg_config_overrides);
+
+    // Build the wkg client
+    let wkg_client = common.get_client_with_config(wkg_client_config).await?;
+
+    // NOTE: Monkey patch & fetch logging performs wkg fetching and placement as a side effect
+    monkey_patch_fetch_logging(wkg_config, dir, &mut lock_file, wkg_client).await?;
+
+    // Now write out the lock file since everything else succeeded
+    lock_file.write().await?;
+
+    Ok("Dependencies fetched".into())
+}
+
+/// Enable extended pull configurations for wkg config
+async fn resolve_extended_pull_configs(
+    pull_cfg: &RegistryPullConfig,
+    wasmcloud_toml_dir: impl AsRef<Path>,
+    wkg_config_overrides: &mut HashMap<String, Override>,
+    wkg_client_config: &mut wasm_pkg_client::Config,
+) -> Result<()> {
+    for RegistryPullSourceOverride { target, source } in pull_cfg.sources.iter() {
+        let (ns, pkgs, _, _, maybe_version) = parse_wit_package_name(target)?;
+        let version_suffix = maybe_version.map(|v| format!("@{v}")).unwrap_or_default();
+
+        match source {
+            // Local files can be used by adding them to the config
+            RegistryPullSource::LocalPath(_) => {
+                let path = source
+                    .resolve_file_path(wasmcloud_toml_dir.as_ref())
+                    .await?;
+                if !tokio::fs::try_exists(&path).await.with_context(|| {
+                    format!(
+                        "failed to check for registry pull source local path [{}]",
+                        path.display()
+                    )
+                })? {
+                    bail!(
+                        "registry pull source path [{}] does not exist",
+                        path.display()
+                    );
+                }
+
+                match (ns, pkgs.as_slice()) {
+                    // namespace-level override
+                    (ns, []) => {
+                        wkg_config_overrides.insert(
+                            ns,
+                            Override {
+                                path: Some(path),
+                                version: None,
+                            },
+                        );
+                    }
+                    // package-level override
+                    (ns, packages) => {
+                        for pkg in packages {
+                            wkg_config_overrides.insert(
+                                format!("{ns}:{pkg}"),
+                                Override {
+                                    path: Some(path.clone()),
+                                    version: None,
+                                },
+                            );
+                        }
+                    }
+                }
```

Comment:
Super nit: This same block is used several times copy pasted. Feels like some of it could be abstracted out into a helper function

Reply:
Sure! The first 3 uses are similar but the last 1 is not actually the same -- right on the border. Will update the first 3